website/static/js/pages/wiki-edit-page.js

```diff
@@ -1,16 +1,153 @@
+var previewElement = $('#viewPreview');
+var versionElement = $('#viewVersion');
+var markdownElement = $('#markdownItRender');
+var compareElement = $('#viewCompare');
+
+var editable = (ctx.panelsUsed.indexOf('edit') !== -1);
+var viewable = (ctx.panelsUsed.indexOf('view') !== -1);
+var comparable = (ctx.panelsUsed.indexOf('compare') !== -1);
+
+var wikiPageOptions = {
+    editVisible: editable,
+    viewVisible: viewable,
+    compareVisible: comparable,
+    canEdit: ctx.canEdit,
+    viewVersion: 'current',
+    compareVersion: 'current',
+    urls: ctx.urls,
+    metadata: ctx.metadata
+};
+
+var wikiPage = new WikiPage('#wikiPageContext', wikiPageOptions);
+
+
+// Edit wiki page name
+if (ctx.canEditPageName) {
+    // Initialize editable wiki page name
+    var $pageName = $('#pageName');
+    $.fn.editable.defaults.mode = 'inline';
+    $pageName.editable({
+        type: 'text',
+        send: 'always',
+        url: ctx.urls.rename,
+        ajaxOptions: {
+            type: 'put',
+            contentType: 'application/json',
+            dataType: 'json'
+        },
+        validate: function(value) {
+            if($.trim(value) === ''){
+                return 'The wiki page name cannot be empty.';
+            } else if(value.length > 100){
+                return 'The wiki page name cannot be more than 100 characters.';
+            }
+        },
+        params: function(params) {
+            return JSON.stringify(params);
+        },
+        success: function(response, value) {
+            window.location.href = ctx.urls.base + encodeURIComponent(value) + '/';
+        },
+        error: function(response) {
+            var msg = response.responseJSON.message_long;
+            if (msg) {
+                return msg;
+            } else {
+                // Log unexpected error with Raven
+                Raven.captureMessage('Error in renaming wiki', {
+                    url: ctx.urls.rename,
+                    responseText: response.responseText,
+                    statusText: response.statusText
+                });
+                return 'An unexpected error occurred. Please try again.';
+            }
+        }
+    });
+}
+
+// Apply panels
+$(document).ready(function () {
+    $('*[data-osf-panel]').osfPanel({
+        buttonElement : '.switch',
+        onSize : 'md',
+        'onclick' : function (title, thisbtn, event ) {
+            // this = all the column elements; an array
+            // title = Text of the button
+            // thisbtn = $(this);
+            // thisbtn.hasClass('btn-primary')
 
-ShareJSDoc('.wiki', url, metadata);
\ No newline at end of file
+            $('body').trigger('togglePanel', [
+                title.toLowerCase(),
+                thisbtn.hasClass('btn-primary')
+            ]);
+
+            if (typeof editor !== 'undefined') { ace.edit(editor).resize(); }
+
+        }
+    });
+    $('.openNewWiki').click(function () {
```

Comment:
Is there a reason to prefer the imperative jQuery interface for modals rather than using `data-target="newWiki"`? I find the latter more concise and robust (one less CSS selector to depend on).